fineract-provider/src/test/java/org/apache/fineract/accounting/productaccountmapping/service/ProductToGLAccountMappingWritePlatformServiceTest.java

```diff
@@ -0,0 +1,124 @@
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.fineract.accounting.productaccountmapping.service;
+
+import static org.mockito.Mockito.any;
+import static org.mockito.Mockito.anyInt;
+import static org.mockito.Mockito.anyLong;
+import static org.mockito.Mockito.anyString;
+import static org.mockito.Mockito.doAnswer;
+import static org.mockito.Mockito.eq;
+import static org.mockito.Mockito.never;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+import java.util.Locale;
+import org.apache.fineract.accounting.common.AccountingConstants.LoanProductAccountingParams;
+import org.apache.fineract.accounting.common.AccountingRuleType;
+import org.apache.fineract.accounting.glaccount.domain.GLAccountType;
+import org.apache.fineract.accounting.producttoaccountmapping.domain.ProductToGLAccountMappingRepository;
+import org.apache.fineract.accounting.producttoaccountmapping.serialization.ProductToGLAccountMappingFromApiJsonDeserializer;
+import org.apache.fineract.accounting.producttoaccountmapping.service.SavingsProductToGLAccountMappingHelper;
+import org.apache.fineract.accounting.producttoaccountmapping.service.ShareProductToGLAccountMappingHelper;
+import org.apache.fineract.infrastructure.core.api.JsonCommand;
+import org.apache.fineract.infrastructure.core.serialization.FromJsonHelper;
+import org.apache.fineract.portfolio.PortfolioProductType;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.mockito.Mock;
+import org.mockito.MockitoAnnotations;
+
+class ProductToGLAccountMappingWritePlatformServiceImplTest {
+
+    @Mock
+    private ProductToGLAccountMappingRepository glAccountMappingRepository;
+
+    @Mock
+    private LoanProductToGLAccountMappingHelper loanProductToGLAccountMappingHelper;
+
+    @Mock
+    private ProductToGLAccountMappingFromApiJsonDeserializer deserializer;
+
+    @Mock
+    private SavingsProductToGLAccountMappingHelper savingsProductToGLAccountMappingHelper;
+
+    @Mock
+    private ShareProductToGLAccountMappingHelper shareProductToGLAccountMappingHelper;
+
+    @Mock
+    private FromJsonHelper fromApiJsonHelper;
+
+    @Mock
+    private JsonCommand command;
+
+    @Mock
+    private JsonElement element;
+
+    private ProductToGLAccountMappingWritePlatformServiceImpl service;
+
+    private static final Long LOAN_PRODUCT_ID = 1L;
+    private static final int PRODUCT_TYPE = PortfolioProductType.LOAN.getValue();
+    private static final int FINANCIAL_ACCOUNT_TYPE_EXPENSE = GLAccountType.EXPENSE.getValue();
+    private static final String CHARGE_OFF_FRAUD_EXPENSE = LoanProductAccountingParams.CHARGE_OFF_FRAUD_EXPENSE.getValue();
+
+    @BeforeEach
```

Comment:
I dont see any added value of having classes that are fully mocked out... I would rather see integration test that is testing whether incoming request got saved properly, and we can fetch created mappings. It can be updated, etc.